### PR TITLE
JOBSBATCH-110 Move Job Batches api to V2

### PR DESCRIPTION
**Pull request by dkBrazz**

### CODE REVIEW

**Comment by dimitrystd** [comment]
@hgronberg-smartling Please don't merge it even if it's reviewed. We would like to get feedback from 2 early adopters and then make an official public release.

File: spec/openapi.yaml [hunk 1]
```diff
@@ -156,25 +156,30 @@ tags:
...
+      up to the Job authorization. As Job Batches Service utilizes internal
       Smartling events it gets more effective than just polling files and jobs.
 
       From the client side flow consists of simple steps:
 
       1. Client creates Job (or uses existing one, which isn't currently
       processed with another Batch)
 
-      2. Client Creates a new Batch for this Job
+      2. Client Creates a new Batch for this Job with the declared list of
```
Comment (dimitrystd) [inline spec/openapi.yaml]: Maybe
```
Client creates a new Batch for this Job with a list of file URIs that will be uploaded into the Job
```

File: spec/openapi.yaml [hunk 2]
```diff
@@ -156,25 +156,30 @@ tags:
...
+      files to be uploaded into the Job
 
-      3. Client adds (uploads) files into the batch giving target locales list
+      3. Client uploads files into the batch giving target locales list
       for each file
 
-      4. When all files are uploaded, client triggers "execute" for the batch.
+      4. Client can cancel some of declared files if there is no reason
```
Comment (dimitrystd) [inline spec/openapi.yaml]: Maybe
```
If one of the declared files cannot be uploaded (for whatever reason) then such file upload can be canceled. An example, a file has no translatable content or there is some error with file generation. In this case, client can cancel file upload and unblock Batch execution
```

File: spec/openapi.yaml [hunk 3]
```diff
@@ -156,25 +156,30 @@ tags:
...
+      in uploading (no translatable content etc.)
 
-      5. Batch execution process waiting until all files are uploaded and added
-      to the Job and then, if was requested, authorizes the Job.
+      5. When all files are uploaded or canceled, "execute" is automatically
```
Comment (dimitrystd) [inline spec/openapi.yaml]:
```
When all files from the declared list are uploaded or canceled...
```

File: spec/openapi.yaml [hunk 4]
```diff
@@ -156,25 +156,30 @@ tags:
...
+      triggered for the batch.
 
-      6. Clients can check batch status and see if all files uploaded and
+      6. Batch execution process waiting until all asynchronous processes with
```
Comment (dimitrystd) [inline spec/openapi.yaml]:
```
Batch execution process waits ...
```

File: spec/openapi.yaml [hunk 5]
```diff
@@ -9691,77 +9462,106 @@ components:
...
+        authorize:
+          description: >-
+            if set true then job will be authorized when upload and attach
+            processes completed for all files in the batch.
+          type: boolean
+        translationJobUid:
+          description: The UID of the previously created translation job.
+          type: string
+        fileUris:
+          description: A list of File Uris to be uploaded later
```
Comment (dimitrystd) [inline spec/openapi.yaml]: Maybe `URIs`

**Review by PavelLoparev — APPROVED** [review]

**Comment by dimitrystd** [comment]
@rachelsmartling Can you pls review and publish?

File: spec/openapi.yaml [hunk 6]
```diff
@@ -142,7 +142,7 @@ tags:
       job, and cancel a job, among many other actions. You have full control via
       the API to manage your job(s).
 
-  - name: Job batches
+  - name: Job Batches V1
```
Comment (rachelsmartling) [inline spec/openapi.yaml:145]: sdfghjk

File: spec/openapi.yaml [hunk 7]
```diff
@@ -189,6 +188,58 @@ tags:
...
       are modified in UI or with another integration.
 
+  - name: Job Batches V2
+    description: |-
+      Job batches take care of asynchronous processes when uploading files and
+      attaching them to jobs.
+
+      Generally, adding multiple files to a job is associated with polling -
+      file is parsed asynchronously and can't be added to a job until it is
+      completed, adding file to a job is also asynchronous, and job can't be
```
Comment (rachelsmartling) [inline spec/openapi.yaml:198]:
```suggestion
      completed. Adding file to a job is also asynchronous, and job can't be
```

File: spec/openapi.yaml [hunk 8]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      authorized until all background tasks completed.
+
+      This forces third-party integration poll every file after every step.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:201]:
```suggestion
      This forces a third-party integration poll for every file after every step.
```

File: spec/openapi.yaml [hunk 9]
```diff
@@ -189,6 +188,58 @@ tags:
...
+
+      Batches hide these asynchronous processes from 3rd party integration -
+      after it accepted file from client it conducts this file through all steps
```
Comment (rachelsmartling) [inline spec/openapi.yaml:204]:
```suggestion
      after it accepts the file from the Smartling User, it processes this file through all steps
```

File: spec/openapi.yaml [hunk 10]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      up to the Job authorization. As Job Batches Service utilizes internal
+      Smartling events it gets more effective than just polling files and jobs.
+
+      From the client side flow consists of simple steps:
```
Comment (rachelsmartling) [inline spec/openapi.yaml:208]:
```suggestion
      From the Smartling user's perspective, the flow consists of the following simple steps:
```

File: spec/openapi.yaml [hunk 11]
```diff
@@ -189,6 +188,58 @@ tags:
...
+
+      1. Client creates Job (or uses existing one, which isn't currently
```
Comment (rachelsmartling) [inline spec/openapi.yaml:210]:
```suggestion
      1. Create a Job (or uses existing one, which isn't currently
```

File: spec/openapi.yaml [hunk 12]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      processed with another Batch)
+
+      2. Client Creates a new Batch for this Job with the declared list of
```
Comment (rachelsmartling) [inline spec/openapi.yaml:213]:
```suggestion
      2. Create a new Batch for this Job with the declared list of
```

File: spec/openapi.yaml [hunk 13]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      files to be uploaded into the Job
+
+      3. Client uploads files into the batch giving target locales list
```
Comment (rachelsmartling) [inline spec/openapi.yaml:216]:
```suggestion
      3. Upload the files into the batch giving target locales list
```

File: spec/openapi.yaml [hunk 14]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      for each file
+
+      4. Client can cancel some of declared files if there is no reason
```
Comment (rachelsmartling) [inline spec/openapi.yaml:219]:
```suggestion
      4. You can cancel some of the declared files if there is no reason
```

File: spec/openapi.yaml [hunk 15]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      in uploading (no translatable content etc.)
+
+      5. When all files are uploaded or canceled, "execute" is automatically
+      triggered for the batch.
+
+      6. Batch execution process waiting until all asynchronous processes with
```
Comment (rachelsmartling) [inline spec/openapi.yaml:225]:
```suggestion
      6. Batch execution process waits until all asynchronous processes
```

File: spec/openapi.yaml [hunk 16]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      uploading and adding to the Job finished and then, if was requested, authorizes the Job.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:226]:
```suggestion
      are complete and then, if requested, authorizes the Job for translation.
```

File: spec/openapi.yaml [hunk 17]
```diff
@@ -189,6 +188,58 @@ tags:
...
+
+      7. Clients can check batch status and see if all files uploaded and
```
Comment (rachelsmartling) [inline spec/openapi.yaml:228]:
```suggestion
      7. You can check batch status and see if all files have been uploaded and
```

File: spec/openapi.yaml [hunk 18]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      attached successfully, what errors happened for each file, and statistics
```
Comment (rachelsmartling) [inline spec/openapi.yaml:229]:
```suggestion
      attached successfully. You can also see if any errors occurred for each file upload and the statistics
```

File: spec/openapi.yaml [hunk 19]
```diff
@@ -189,6 +188,58 @@ tags:
...
+      for added/skipped strings.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:230]:
```suggestion
      for ingested/excluded strings.
```

File: spec/openapi.yaml [hunk 20]
```diff
@@ -8637,7 +8709,7 @@ paths:
 
 
                     **Note**: File will not be authorized for these locales once
-                    it is uploaded. Authorization is happened only when job is
+                    it is uploaded. Authorization is happened only when a job is
```
Comment (rachelsmartling) [inline spec/openapi.yaml:8712]:
```suggestion
                    it is uploaded. Authorization happens only when a job is
```

File: spec/openapi.yaml [hunk 21]
```diff
@@ -8720,20 +8792,560 @@ paths:
...
+  #
+  # Job Batches V2
+  #
+
+  /job-batches-api/v2/projects/{projectId}/batches:
+    post:
+      summary: Create batch
+      description: >-
+        Creates a new batch for the given Job with a list of file names to be uploaded.
+        File names can be added during files uploading to the batch, but only before the batch will start executing.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:8847]:
```suggestion
        File names can be added while files are uploading to the batch, but only before the batch will start executing.
```

File: spec/openapi.yaml [hunk 22]
```diff
@@ -8720,20 +8792,560 @@ paths:
...
+        The batch is executed once the last file in the initial list or added files is uploaded or cancelled
```
Comment (rachelsmartling) [inline spec/openapi.yaml:8848]:
```suggestion
        The batch is executed as soon as the last file in the initial list or added files are uploaded or cancelled
```

File: spec/openapi.yaml [hunk 23]
```diff
@@ -8720,20 +8792,560 @@ paths:
...
+                    Upload` but it doesn't relay it to File Api
+                  type: boolean
+                'localeIdsToAuthorize[]':
+                  description: >
+                    This value declares locales list which should be used when
+                    file is attached to a job.
+
+
+                    **Note**: File will not be authorized for these locales once
+                    it is uploaded. Authorization is happened only when a job is
```
Comment (rachelsmartling) [inline spec/openapi.yaml:9259]:
```suggestion
                    it is uploaded. Authorization happens only when a job is
```

File: spec/openapi.yaml [hunk 24]
```diff
@@ -9691,77 +10304,184 @@ components:
           description: Number of strings not added to the translationJob from this file
           type: integer
       type: object
-    JobBatchItemStatusResponse:
+    BatchItemLocaleV2:
+      description: A smartling locale which file should be authorized to
```
Comment (rachelsmartling) [inline spec/openapi.yaml:10308]:
```suggestion
      description: A smartling locale which the file should be authorized and translated into
```

File: spec/openapi.yaml [hunk 25]
```diff
@@ -9691,77 +10304,184 @@ components:
...
+        stringsAdded:
+          description: Number of strings added to the translationJob from this file
+          type: integer
+      type: object
+    CreateBatchRequestV1:
+      description: The payload to use when creating a job batch.
+      properties:
+        authorize:
+          description: >-
+            if set true then job will be authorized when upload and attach
```
Comment (rachelsmartling) [inline spec/openapi.yaml:10322]:
```suggestion
            if the set is true then the job will be authorized when uploaded and attach
```

File: spec/openapi.yaml [hunk 26]
```diff
@@ -9691,77 +10304,184 @@ components:
...
+        translationJobUid:
+          description: The UID of the previously created translation job.
+          type: string
+      type: object
+    CreateBatchRequestV2:
+      description: The payload to use when creating a job batch.
+      properties:
+        authorize:
+          description: >-
+            if set true then job will be authorized when upload and attach
```
Comment (rachelsmartling) [inline spec/openapi.yaml:10334]:
```suggestion
            if the set is true then the job will be authorized when uploaded and attach
```

File: spec/openapi.yaml [hunk 27]
```diff
@@ -9691,77 +10304,184 @@ components:
...
+            processes completed for all files in the batch.
+          type: boolean
+        translationJobUid:
+          description: The UID of the previously created translation job.
+          type: string
+        fileUris:
+          description: A list of File Uris to be uploaded later
```
Comment (rachelsmartling) [inline spec/openapi.yaml:10341]:
```suggestion
          description: A list of File URI's to be uploaded later
```

**Review by rachelsmartling — CHANGES_REQUESTED** [review]

**Pull request merged**